### PR TITLE
PHOENIX-6442 Phoenix should depend on the appropriate tephra-hbase-co…

### PR DESCRIPTION
**Pull request by stoty**
…mpat-x.y module

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1611,6 +1615,7 @@
         <hbase.version>${hbase-2.4.0.runtime.version}</hbase.version>
         <hbase.thirdparty.version>3.4.1</hbase.thirdparty.version>
         <zookeeper.version>3.5.7</zookeeper.version>
+        <tephra.hbase.compat.version>2.4</tephra.hbase.compat.version>
```
Comment (virajjasani) [inline pom.xml:1618]: Tephra module doesn't have any compat issues b/ HBase 2.4.0 and 2.4.1 (the way Phoenix has) right?

**Review by virajjasani — COMMENTED** [review]
Looks good, pending QA

File: pom.xml (hunk 1 above)
Comment (stoty) [reply pom.xml:1618]: Yes, that's my understanding.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 38s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  15m 50s |  master passed  |
| +0 |  hbaserecompile  |  30m 48s |  HBase recompiled.  |
| +1 :green_heart: |  compile  |   2m  8s |  master passed  |
| +1 :green_heart: |  javadoc  |   3m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 36s |  root in the patch failed.  |
| +0 |  hbaserecompile  |  11m 48s |  HBase recompiled.  |
| -1 :x: |  compile  |   0m 33s |  root in the patch failed.  |
| -1 :x: |  javac  |   0m 33s |  root in the patch failed.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 36s |  root in the patch failed.  |
| -1 :x: |  javadoc  |   0m 14s |  phoenix-core in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 37s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 38s |  The patch generated 613 ASF License warnings.  |
|  |   |  54m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1208 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux f71b665d1ecd 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / d1f26c7 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/patch-compile-root.txt |
| javac | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/patch-compile-root.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/patch-javadoc-root.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/patch-javadoc-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: . phoenix-core U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
This is normal, @virajjasani 
This patch needs tephra 0.16.1

**Comment by virajjasani** [comment]
> This patch needs tephra 0.16.1

Is it 0.16.1 or 0.16.0? The error says `Failure to find org.apache.tephra:tephra-hbase-compat-2.4:jar:shaded:0.16.0`. I think I am missing something.

**Comment by virajjasani** [comment]
For instance, https://repository.apache.org/content/repositories/releases/org/apache/tephra/tephra-hbase-compat-2.3/ has 0.16.0, and since 2.4 compat module doesn't exist, it will fail but it should fail for 0.16.0 only since we haven't yet updated tephra version right? I think this looks good, I got it.

**Review by virajjasani — APPROVED** [review]

**Comment by virajjasani** [comment]
> This patch needs tephra 0.16.1

Oh, btw did you mean that this error should be resolved after 0.16.1 is rolled out?

**Comment by virajjasani** [comment]
Sorry for all confusions, I was into 2 things simultaneously. I can start off with 0.16.1 RC preparation and can test RC locally with this PR changes + tephra version update in pom.xml.
Thanks @stoty

**Comment by virajjasani** [comment]
@stoty you might also want to bump `tephra.version` to 0.16.1. I just tested this PR locally and it's working fine so far.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m  0s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m  3s |  master passed  |
| +0 |  hbaserecompile  |  22m 39s |  HBase recompiled.  |
| +1 :green_heart: |  compile  |   1m 32s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 13s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   8m 20s |  the patch passed  |
| +0 |  hbaserecompile  |  14m 48s |  HBase recompiled.  |
| +1 :green_heart: |  compile  |   1m 33s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 33s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   2m 14s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 113m  4s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   1m 46s |  The patch generated 613 ASF License warnings.  |
|  |   | 163m 34s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.PermissionNSEnabledWithCustomAccessControllerIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1208 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux a86e1813994a 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 877457e |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/3/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/3/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/3/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 15580 (vs. ulimit of 30000) |
| modules | C: phoenix-core . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1208/3/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**